src/app/shared/dynamic-form/components/form-lazy-options-base.component.ts

```diff
@@ -96,40 +118,25 @@ export class FormLazyOptionsBaseComponent extends FormBaseComponent implements O
     );
   }
 
-  makeFiltersFromValue(val: any): any[] | null {
-    const key = this.config.determinedBy;
-    if (!key) {
+  makeFilters(): Filter[] | null {
+    if (!this.config.determinedBy?.length) {
       return null;
     }
-    const field = this.makeKey(key.replaceAll('.', '__'));
-    const filters = Object.entries(this.makeObject(field, val))[0];
-
-    const keyMap = ['country', 'region', 'district', 'community'];
-    const filterName = keyMap.find((item) => key.toLowerCase().includes(item));
-
-    return [
-      {
-        field: filterName,
-        value: filters[1],
-      },
-    ];
-  }
-
-  makeKey(key: string): string {
-    const keyMap = ['country', 'region', 'district', 'community'];
-    const test = keyMap.find((item) => key.toLowerCase().includes(item));
-    if (!test) {
-      return key;
-    }
-    const index = key.toLowerCase().indexOf(test);
-    let result = key.substr(index);
-    result = result[0].toLowerCase() + result.slice(1);
-    return result;
+    return Object.entries(this.determinantValues)
+      .filter(([, val]) => !!val)
+      .map(([key, val]) => {
+        const dependency = this.config.determinedBy?.find((item) => item.key === key);
+        const [field, value] = Object.entries(this.makeObject(dependency?.keyMap || key, val))[0];
```

Comment:
will there always be an element in this array? index = 0

Reply:
no, Object.entries could return an empty array
but that won't break the code, field and value would just be undefined (and ignored as filters)